File: src/models/Score.py
from .. import mongodb
from ..common.utils import generate_uuid, time_now


class Score(mongodb.DynamicDocument):
    uuid = mongodb.UUIDField(binary=False, default=generate_uuid)
    ctime = mongodb.IntField(default=time_now)
    mtime = mongodb.IntField(onupdate=time_now)
    contest_uuid = mongodb.UUIDField(binary=False, required=True)
    status = mongodb.StringField(required=True)

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

File: src/models/score.ts
/**
 * External dependencies
 */
import * as mongoose from 'mongoose';
import { generateUUID } from '../common/utils';
/**
 * Create the score schema
 */
const score: mongoose.Schema = new mongoose.Schema(
    {
        uuid: {
            type: String,
            required: true,
            default: generateUUID,
        },
        contest_uuid: {
            type: String,
            required: true,
        },
        status: {
            type: String,
            required: true,
        },
        sheet: {
            type: Array,
            required: false,
        },
    },
    {
        timestamps: {
            createdAt: 'ctime',
            updatedAt: 'mtime',
        },
    }
);
/**
 * Export the model
 */
export default mongoose.model('Score', score);
